plugins/amqp_trx_plugin/amqp_trx_plugin.cpp

```diff
@@ -187,15 +192,24 @@ void amqp_trx_plugin::set_program_options(options_description& cli, options_desc
    op("amqp-trx-ack-mode", bpo::value<ack_mode>()->default_value(ack_mode::in_block),
       "AMQP ack when 'received' from AMQP, when 'executed', or when 'in_block' is produced that contains trx.\n"
       "Options: received, executed, in_block");
+   op("amqp-trx-trace-address", bpo::value<std::string>(),
+      "AMQP address: Format: amqp://USER:PASSWORD@ADDRESS:PORT\n"
+      "Will publish to amqp-trx-trace-queue-name ('trace') queue. If not provided traces are not sent.");
+   op("amqp-trx-trace-queue-name", bpo::value<std::string>()->default_value("trace"),
+      "AMQP queue to publish transaction traces of amqp transactions.");
+   op("amqp-trx-trace-exchange", bpo::value<std::string>()->default_value(""),
+      "Existing AMQP exchange to send transaction trace messages.");
```

Comment:
I don't think that these should be options.  The trx plugin should _only_ send a trace via `reply-to`, it should _always_ send a trace when a `reply-to` is given, and it should always send the trace to the same AMQP server from which it receives transactions.  Broadcasting traces is the responsibility of the trace plugin.

Reply:
This is a really good point. @arhag do you agree with this?